src/python/CRABClient/UserUtilities.py

```diff
@@ -114,15 +114,22 @@ def getUsernameFromSiteDB():
     return username
 
 
-def getFileFromURL(url, filename = None):
+def getFileFromURL(url, filename = None, proxyfilename = None):
     """
     Read the content of a URL and copy it into a file.
+
+    url: the link you would like to retrieve
+    filename: the local filename where the url is saved to. Defaults to the filename in the url
+    proxyfilename: the x509 proxy certificate to be used in case auth is required
+
+    Return the filename used to save the file or ClientException in case of errors.
     """
     if filename == None:
         path = urlparse(url).path
         filename = os.path.basename(path)
     try:
-        socket = urllib.urlopen(url)
+        opener = urllib.URLopener(key_file = proxyfilename, cert_file = proxyfilename)
+        socket = opener.open(url)
         status = socket.getcode()
```

Comment:
Do we want to support no HTTP URL? Because in that case getcode() returns `None`, which means we would have to change https://github.com/dmwm/CRABClient/pull/4554/files#diff-52c590cea8f33d9929a472a7768244a6R146